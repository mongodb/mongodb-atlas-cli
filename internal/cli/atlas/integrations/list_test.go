// Copyright 2020 MongoDB Inc
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

//go:build unit

package integrations

import (
	"bytes"
	"testing"

	"github.com/golang/mock/gomock"
	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
	"github.com/mongodb/mongodb-atlas-cli/internal/mocks"
	"github.com/mongodb/mongodb-atlas-cli/internal/pointer"
	"github.com/mongodb/mongodb-atlas-cli/internal/test"
	"github.com/stretchr/testify/assert"
	atlasv2 "go.mongodb.org/atlas-sdk/admin"
)

func TestDBUserList_Run(t *testing.T) {
	ctrl := gomock.NewController(t)
	mockStore := mocks.NewMockIntegrationLister(ctrl)

	buf := new(bytes.Buffer)

	listOpts := &ListOpts{
		store: mockStore,
		OutputOpts: cli.OutputOpts{
			Template:  listTemplate,
			OutWriter: buf,
		},
	}

	expected := &atlasv2.PaginatedIntegration{
		Results: []atlasv2.ThridPartyIntegration{
			{
				Datadog: &atlasv2.Datadog{
					Type:   pointer.Get("DATADOG"),
					ApiKey: "testsApiKey",
				},
			},
			{
				MicrosoftTeams: &atlasv2.MicrosoftTeams{
					Type:                     pointer.Get("MICROSOFT_TEAMS"),
					MicrosoftTeamsWebhookUrl: "urlHook",
				},
			},
		},
	}

	mockStore.
		EXPECT().
		Integrations(listOpts.ProjectID).
		Return(expected, nil).
		Times(1)

	if err := listOpts.Run(); err != nil {
		t.Fatalf("Run() unexpected error: %v", err)
	}

	assert.Equal(t, `TYPE
DATADOG
MICROSOFT_TEAMS
`, buf.String())
	t.Log(buf.String())
}

func TestListBuilder(t *testing.T) {
	test.CmdValidator(
		t,
		ListBuilder(),
		0,
		[]string{flag.ProjectID},
	)
}
