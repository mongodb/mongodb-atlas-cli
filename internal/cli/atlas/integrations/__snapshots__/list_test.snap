
[TestDBUserList_Run - 1]
TYPE
DATADOG
MICROSOFT_TEAMS

---
